[Failing Test]: Python PostCommit perma-red due to Flink Python xlang test failures: "apache/beam_java11_sdk:2.61.0.dev not found"

### What happened?

First failure: 

https://github.com/apache/beam/actions/runs/11445520520

https://github.com/apache/beam/runs/31848196342

```
      if self._runtime_exception:
>       raise self._runtime_exception
E       RuntimeError: Pipeline BeamApp-runner-1021175329-11c8c557_04f95baf-bc86-48dc-954b-16138b058df3 failed in state FAILED: java.io.IOException: Received exit code 125 for command 'docker run -d --network=host --env=DOCKER_MAC_CONTAINER=null apache/beam_java11_sdk:2.61.0.dev --id=1-2 --provision_endpoint=localhost:41031'. stderr: Unable to find image 'apache/beam_java11_sdk:2.61.0.dev' locallydocker: Error response from daemon: manifest for apache/beam_java11_sdk:2.61.0.dev not found: manifest unknown: manifest unknown.See 'docker run --help'.

apache_beam/runners/portability/portable_runner.py:568: RuntimeError
```



### Issue Failure

Failure: Test is flaky

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [X] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/.github/trigger_files/beam_PostCommit_Python.json
+++ b/.github/trigger_files/beam_PostCommit_Python.json
@@ -1,5 +1,5 @@
 {
   "comment": "Modify this file in a trivial way to cause this test suite to run.",
-  "modification": 4
+  "modification": 2
 }
 

--- a/sdks/python/test-suites/portable/common.gradle
+++ b/sdks/python/test-suites/portable/common.gradle
@@ -23,6 +23,7 @@ import org.apache.tools.ant.taskdefs.condition.Os
 def pythonRootDir = "${rootDir}/sdks/python"
 def pythonVersionSuffix = project.ext.pythonVersion.replace('.', '')
 def latestFlinkVersion = project.ext.latestFlinkVersion
+def currentJavaVersion = project.ext.currentJavaVersion
 
 ext {
   pythonContainerTask = ":sdks:python:container:py${pythonVersionSuffix}:docker"
@@ -369,7 +370,7 @@ project.tasks.register("postCommitPy${pythonVersionSuffix}IT") {
           'setupVirtualenv',
           'installGcpTest',
           ":runners:flink:${latestFlinkVersion}:job-server:shadowJar",
-          ':sdks:java:container:java8:docker',
+          ":sdks:java:container:${currentJavaVersion}:docker",
           ':sdks:java:testing:kafka-service:buildTestKafkaServiceJar',
           ':sdks:java:io:expansion-service:shadowJar',
           ':sdks:java:io:google-cloud-platform:expansion-service:shadowJar',
@@ -420,7 +421,7 @@ project.tasks.register("xlangSpannerIOIT") {
           'setupVirtualenv',
           'installGcpTest',
           ":runners:flink:${latestFlinkVersion}:job-server:shadowJar",
-          ':sdks:java:container:java8:docker',
+          ":sdks:java:container:${currentJavaVersion}:docker",
           ':sdks:java:io:expansion-service:shadowJar',
           ':sdks:java:io:google-cloud-platform:expansion-service:shadowJar',
           ':sdks:java:io:kinesis:expansion-service:shadowJar',

--- a/website/www/site/content/en/documentation/runtime/environments.md
+++ b/website/www/site/content/en/documentation/runtime/environments.md
@@ -111,14 +111,14 @@ This method requires building image artifacts from Beam source. For additional i
   cd $BEAM_WORKDIR
 
   # The default repository of each SDK
-  ./gradlew :sdks:java:container:java8:docker
   ./gradlew :sdks:java:container:java11:docker
   ./gradlew :sdks:java:container:java17:docker
+  ./gradlew :sdks:java:container:java21:docker
   ./gradlew :sdks:go:container:docker
-  ./gradlew :sdks:python:container:py38:docker
   ./gradlew :sdks:python:container:py39:docker
   ./gradlew :sdks:python:container:py310:docker
   ./gradlew :sdks:python:container:py311:docker
+  ./gradlew :sdks:python:container:py312:docker
 
   # Shortcut for building all Python SDKs
   ./gradlew :sdks:python:container:buildAll
@@ -168,9 +168,9 @@ builds the Python 3.6 container and tags it as `example-repo/beam_python3.6_sdk:
 From Beam 2.21.0 and later, a `docker-pull-licenses` flag was introduced to add licenses/notices for third party dependencies to the docker images. For example:
 
 ```
-./gradlew :sdks:java:container:java8:docker -Pdocker-pull-licenses
+./gradlew :sdks:java:container:java11:docker -Pdocker-pull-licenses
 ```
-creates a Java 8 SDK image with appropriate licenses in `/opt/apache/beam/third_party_licenses/`.
+creates a Java 11 SDK image with appropriate licenses in `/opt/apache/beam/third_party_licenses/`.
 
 By default, no licenses/notices are added to the docker images.
 

